pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -570,29 +573,78 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*corev1.Secret, error) {
-	var secrets []*corev1.Secret
+func (r *ReconcileLogStorage) getCertificateKeySecret(ctx context.Context, keySecretName string, keySecretNs string, pubSecretName string, pubSecretNs string, svcDNSNames ...string) (*corev1.Secret, error) {
```

Comment:
I don't think this function belongs to the logstorage reconciler, could you please put this in a suitable common location.